oauth_gateway.go

```diff
@@ -0,0 +1,39 @@
+package braintree
+
+import (
+	"context"
+	"encoding/xml"
+)
+
+type OauthGateway struct {
+	*Braintree
+}
+
+type OAuthCredentialRequest struct {
+	XMLName      xml.Name `xml:"credentials"`
+	Code         string   `xml:"code"`
+	Scope        string   `xml:"scope"`
+	GrantType    string   `xml:"grantType"`
+	RefreshToken string   `xml:"refreshToken"`
+}
+
+type OAuthCredentials struct {
```

Comment:
Do we need a XML name here in order to unmarshal correctly?

Reply:
Looking at its Java API, it doesn't have a name for `OAuthCredentials` - same as the existing `PaymentMethodNonce`